bindings/go/helm/input/method_test.go

```diff
@@ -0,0 +1,240 @@
+						assert.Equal(t, "registry.example.com", identity["hostname"])
+						assert.Equal(t, "8443", identity["port"])
+						assert.Equal(t, "helm/charts/myapp-1.0.0.tgz", identity["path"])
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestInputMethod_ProcessResource_LocalChart(t *testing.T) {
+	testDataDir := filepath.Join("testdata", "mychart")
+	helmSpec := v1.Helm{
+		Type: runtime.Type{
+			Name: v1.Type,
+		},
+		Path: testDataDir,
+	}
+
+	resource := &constructorruntime.Resource{
+		AccessOrInput: constructorruntime.AccessOrInput{
+			Input: &helmSpec,
+		},
+	}
+
+	inputMethod := &input.InputMethod{}
+	result, err := inputMethod.ProcessResource(t.Context(), resource, nil)
+
+	require.NoError(t, err)
+	assert.NotNil(t, result)
+	assert.NotNil(t, result.ProcessedBlobData, "should have blob data for local chart")
+	assert.Nil(t, result.ProcessedResource, "should not have remote resource for local chart")
+}
+
+func TestInputMethod_ProcessResource_RemoteChart_Podinfo_Integration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test in short mode")
+	}
+
+	helmSpec := v1.Helm{
+		Type: runtime.Type{
+			Name: v1.Type,
+		},
+		// TODO: Only direct URLs to chart tgz files are supported currently.
+		// Need to add support for index.yaml based repositories maybe?
+		// "https://stefanprodan.github.io/podinfo/" is supposed to be the index URL or rather the repository.
+		// For now, using direct link to a specific chart version.
+		HelmRepository: "https://stefanprodan.github.io/podinfo/podinfo-6.9.1.tgz",
+		Version:        "6.9.1",
+	}
+
+	resource := &constructorruntime.Resource{
+		AccessOrInput: constructorruntime.AccessOrInput{
+			Input: &helmSpec,
+		},
+	}
+
+	inputMethod := &input.InputMethod{}
+
+	result, err := inputMethod.ProcessResource(t.Context(), resource, nil)
+
+	require.NoError(t, err, "should successfully download podinfo chart")
+	assert.NotNil(t, result, "result should not be nil")
+	assert.NotNil(t, result.ProcessedBlobData, "should have blob data for remote chart")
+	assert.NotNil(t, result.ProcessedResource, "should have remote resource access info")
+
+	// Verify the remote resource structure
+	assert.Equal(t, "podinfo-6.9.1.tgz", result.ProcessedResource.Name, "chart name should be extracted correctly")
+	assert.Equal(t, "6.9.1", result.ProcessedResource.Version, "version should match specification")
+	assert.Equal(t, input.HelmRepositoryType, result.ProcessedResource.Type, "resource type should be helmRepository")
+
+	// Verify blob data is not empty by reading some content
+	reader, err := result.ProcessedBlobData.ReadCloser()
+	require.NoError(t, err)
+	defer reader.Close()
+
+	// Read first few bytes to verify content exists
+	buffer := make([]byte, 100)
+	n, err := reader.Read(buffer)
+	require.NoError(t, err)
+	assert.Greater(t, n, 0, "blob should contain data")
+}
+
+func TestInputMethod_ProcessResource_RemoteChart_PodinfoLatest_Integration(t *testing.T) {
```

Comment:
TODO: This actually NOT using latest yet. Needs to be checked to download without defining a version.